src/iuse.cpp

```diff
@@ -3809,6 +3809,60 @@ int iuse::mp3_on(player *p, item *it, bool t, const tripoint &pos)
     return it->type->charges_to_use();
 }
 
+int iuse::gasmask(player *p, item *it, bool, const tripoint& )
```

Comment:
This should be implemented as `iuse_actor` instead. It allows to make the `oname` a property of the action, not a calculated value.

Reply:
You probably don't even need two functions:
- you can detect activeness via the item property,
- detect explicit invocation vs. automatic each turn invocation via the `t` parameter.

See for example `iuse::radiocontrol`.